src/vs/workbench/contrib/debug/browser/loadedScriptsQuickAccess.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { Codicon } from 'vs/base/common/codicons';
+import { matchesFuzzy } from 'vs/base/common/filters';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { IPickerQuickAccessItem, PickerQuickAccessProvider } from 'vs/platform/quickinput/browser/pickerQuickAccess';
+import { IQuickPickSeparator } from 'vs/platform/quickinput/common/quickInput';
+import { LOADED_SCRIPTS_QUICK_PICK_PREFIX } from 'vs/workbench/contrib/debug/browser/debugCommands';
+import { IDebugService, IDebugSession } from 'vs/workbench/contrib/debug/common/debug';
+import { Source } from 'vs/workbench/contrib/debug/common/debugSource';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+
+export class LoadedScriptQuickAccess extends PickerQuickAccessProvider<IPickerQuickAccessItem> {
+
+	protected _getLabelName(source: Source) {
+		const posOfForwardSlash = source.name.lastIndexOf('/');
+		const posOfBackSlash = source.name.lastIndexOf('\\');
+
+		const delimeterPos = posOfForwardSlash > posOfBackSlash ? posOfForwardSlash : posOfBackSlash;
+		const label = source.name.substring(delimeterPos + 1);
+		return label;
+	}
+	protected _getPicksFromSession(session: IDebugSession, filter: string): Promise<Array<IPickerQuickAccessItem | IQuickPickSeparator>> {
+
+		return new Promise((resolve, reject) => {
```

Comment:
I was creating a promise manually so that I could use `Promise.all` to run all of the async processes at once (to get loaded sources of all sessions in parallel)

Reply:
Async methods just return a normal promise. You don't have to instantiate a promise for that.